crates/red_knot_test/src/parser.rs

```diff
@@ -664,6 +677,18 @@ mod tests {
         assert_eq!(file.code, "x = 10");
     }
 
+    #[test]
+    fn unterminated_code_block() {
+        let source = dedent(
+            "
+            ```
+            x = 1
+            ",
+        );
+        let err = super::parse("file.md", &source).expect_err("Should fail to parse");
+        assert_eq!(err.to_string(), "Unterminated code block at index 0.");
```

Comment:
Maybe use an example where the code block doesn't start at position 0 in the source text (i.e. preface the source text with some prose or a header) so this test would have caught the issue with the index reporting.